src/main.ts

```diff
@@ -182,6 +184,72 @@ export function generateFile(ctx: Context, fileDesc: FileDescriptorProto): [stri
     chunks.push(generateDataLoadersType());
   }
 
+  if (options.outputSchema) {
+    chunks.push(code`
+      export interface ProtoMetadata {
+        fileDescriptor: ${imp('IFileDescriptorProto@protobufjs/ext/descriptor')};
```

Comment:
Nice! Glad this approach worked out.

Fwiw I've been meaning to publish ts-proto-generated versions of `FileDescriptorProto` to npm, like `ts-proto-protobuf-descriptors` or something like that, and then use that for parsing the `protoc` input messages, instead of the current protobufjs types (which I had to use when originally writing ts-proto b/c ts-proto didn't exist yet :-) ).

But, just mentioning that, b/c then these type could be the ts-proto-generated `FileDescriptorProto` would might be slightly more pleasant to use (or at least somewhat different).

We don't have to do that for this PR, but just mentioning it as something that I might get around to doing.

Reply:
That could add some convenience if you are only generating the typescript typings and metadata, however, if you are also generating the json methods or client implementations, it instantly adds the dependency to `protobufjs` which would sort of defeat the purpose. (and I also consider the official typings a bit more stable than ones generated from protos)

Reply:
> you are also generating the json methods or client implementations

True, it wouldn't need the json methods... There would be no client impls b/c the descriptor `*.proto` files only have messages and no services.

Re the `protobufjs` dependency, ts-proto already has this (at build-time) to decode the `protoc` plugin's stdin messages, via the currently-generated-by-pbjs messages, so just switching to ts-proto-generated messages would use the same `protobufjs` dependency for doing the raw binary decoding. So wouldn't/shouldn't be a new dependency.

> official typings a bit more stable than ones generated from protos

AFAIU these protobufjs typings are themselves generated from `*.proto`s, so I don't know that they are any more or less official than what ts-proto would similarly generate from the [descriptor](https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.descriptor.pb) proto file.

All said, I'm fine leaving this to later, it just means the caveat of a slight breaking change if/when `fileDescriptor` moves to use a ts-proto-generated message type.

Reply:
Ok, yeah, that makes sense, and if the official typings are also generated from the protos, then there shouldn't really be any serious breaking changes when transferring over.

Reply:
Fwiw I went ahead and moved over to ts-proto-generated messages for both the metadata in this PR and also just the core stdin/stdout message plugin parsing. I don't think there should be any significant differences but let me know if so.